src/sentry/static/sentry/app/views/releases/detail/overview/chart/utils.tsx

```diff
@@ -40,62 +41,83 @@ export function getReleaseEventView(
+    environment: environments,
+    projects,
+    start: start ? getUtcDateString(start) : undefined,
+    end: end ? getUtcDateString(end) : undefined,
+  };
+
   switch (yAxis) {
-    case YAxis.ALL_TRANSACTIONS:
     case YAxis.FAILED_TRANSACTIONS:
-      const statusFilter =
-        yAxis === YAxis.FAILED_TRANSACTIONS
-          ? ['ok', 'cancelled', 'unknown'].map(s => `!transaction.status:${s}`).join(' ')
-          : '';
+      const statusFilters = ['ok', 'cancelled', 'unknown'].map(
+        s => `!transaction.status:${s}`
+      );
       return EventView.fromSavedQuery({
-        id: undefined,
-        version: 2,
-        name: `${t('Release')} ${formatVersion(version)}`,
-        fields: [`count()`, toOther],
-        query: `${releaseFilter} ${statusFilter}`.trim(),
-        orderby: toOtherAlias,
-        range: period,
-        environment: environments,
-        projects,
-        start: start ? getUtcDateString(start) : undefined,
-        end: end ? getUtcDateString(end) : undefined,
+        ...baseQuery,
+        query: stringifyQueryObject(
+          new QueryResults(
+            ['event.type:transaction', releaseFilter, ...statusFilters].filter(Boolean)
+          )
+        ),
       });
     case YAxis.COUNT_LCP:
     case YAxis.COUNT_DURATION:
       const column =
         yAxis === YAxis.COUNT_DURATION ? 'transaction.duration' : 'measurements.lcp';
       const threshold =
         yAxis === YAxis.COUNT_DURATION
-          ? organization.apdexThreshold
+          ? organization?.apdexThreshold
           : WEB_VITAL_DETAILS[WebVital.LCP].failureThreshold;
       return EventView.fromSavedQuery({
-        id: undefined,
-        version: 2,
-        name: `${t('Release')} ${formatVersion(version)}`,
-        fields: ['count()', toOther],
-        query: `event.type:transaction ${releaseFilter} ${column}:>${threshold}`,
-        orderby: toOtherAlias,
-        range: period,
-        environment: environments,
-        projects,
-        start: start ? getUtcDateString(start) : undefined,
-        end: end ? getUtcDateString(end) : undefined,
+        ...baseQuery,
+        query: stringifyQueryObject(
+          new QueryResults(
+            [
+              'event.type:transaction',
+              releaseFilter,
+              threshold ? `${column}:>${threshold}` : '',
+            ].filter(Boolean)
+          )
+        ),
       });
+    case YAxis.EVENTS:
+      if (
+        organization?.features?.includes('performance-view') &&
+        organization?.features?.includes('release-performance-views')
+      ) {
+        const eventTypeFilter = eventType === 'all' ? '' : `event.type:${eventType}`;
+        return EventView.fromSavedQuery({
+          ...baseQuery,
+          query: stringifyQueryObject(
+            new QueryResults([releaseFilter, eventTypeFilter].filter(Boolean))
+          ),
+        });
+      } else {
+        // TODO(tonyx): Delete this else once the feature flags are removed
```

Comment:
Won't this always depend on the organization having `performance-view` or `discover-basic`?

Reply:
This else block exists to make sure the existing behaviour does not change until we remove the `release-performance-views` feature flag. Without this check, the existing `All Events` y-axis option will immediately begin using the new query.

Reply:
I think removing that `performance-view` feature flag check above would make the intent here clearer. When the selected Y-Axis is `YAxis.Events`, we have 2 possible views.

1. The new view with this release vs other releases which is available with the `release-performance-views` feature flag
2. The existing view of the non-transaction events that we are replacing.